src/app-layout/utils/use-drawers.ts

```diff
@@ -91,7 +101,7 @@ export function useDrawers(
   const [activeDrawerId, setActiveDrawerId] = useControllable(
     ownDrawers?.activeDrawerId,
     ownDrawers?.onChange,
-    !isMobile && !toolsProps.toolsHide && toolsProps.toolsOpen && defaults.toolsOpen ? TOOLS_DRAWER_ID : undefined,
+    getInitialDrawerId(isMobile, toolsProps, defaults),
```

Comment:
There was a logical error in this code. Refactored into separate function to make it easier to read and follow the logic